crates/forge/test_utils/src/runner.rs

```diff
@@ -295,7 +295,86 @@ macro_rules! assert_gas {
                     panic!("Cannot use assert_gas! for fuzzing tests")
                 }
                 AnyTestCaseSummary::Single(case) => match case {
-                    TestCaseSummary::Passed { gas_info: gas, .. } => *gas == $asserted_gas,
+                    TestCaseSummary::Passed { gas_info: gas, .. } => {
+                        *gas == $asserted_gas
+                            && any_case
+                                .name()
+                                .unwrap()
+                                .ends_with(test_name_suffix.as_str())
```

Comment:
Why is this needed? Isn't it filtered already by now?

Reply:
Apparently, it's not. It just worked because there was always a single cairo test in the code.